parlai/mturk/webapp/server.py

```diff
@@ -0,0 +1,375 @@
+
+
+class WorkerListHandler(BaseHandler):
+    def initialize(self, app):
+        self.state = app.state
+        self.subs = app.subs
+        self.sources = app.sources
+        self.port = app.port
+        self.data_handler = app.data_handler
+
+    def get(self):
+        results = self.data_handler.get_all_worker_data()
+        processed_results = []
+        for res in results:
+            processed_results.append(dict(zip(res.keys(), res)))
+
+        self.write(json.dumps(processed_results))
+
+
+class WorkerHandler(BaseHandler):
+    def initialize(self, app):
+        self.state = app.state
+        self.subs = app.subs
+        self.sources = app.sources
+        self.port = app.port
+        self.data_handler = app.data_handler
+
+    def get(self, worker_target):
+        assignments = self.data_handler.get_all_assignments_for_worker(
+            worker_target)
+        pairings = self.data_handler.get_all_pairings_for_worker(
+            worker_target)
+        processed_assignments = merge_assignments_with_pairings(
+            assignments, pairings, 'task {}'.format(worker_target))
+        worker_details = row_to_dict(
+            self.data_handler.get_worker_data(worker_target))
+        data = {
+            'worker_details': worker_details,
+            'assignments': processed_assignments,
+        }
+
+        self.write(json.dumps(data))
+
+class ErrorHandler(BaseHandler):
+    def get(self, text):
+        error_text = text or "test error"
+        raise Exception(error_text)
+
+
+def start_server(port=DEFAULT_PORT, hostname=DEFAULT_HOSTNAME,
+                 db_file=DEFAULT_DB_FILE):
+    print("It's Alive!")
+    app = Application(port=port, db_file=db_file)
+    app.listen(port, max_buffer_size=1024 ** 3)
+    logging.info("Application Started")
+
+    if "HOSTNAME" in os.environ and hostname == DEFAULT_HOSTNAME:
+        hostname = os.environ["HOSTNAME"]
+    else:
+        hostname = hostname
+    print("You can navigate to http://%s:%s" % (hostname, port))
+    tornado.ioloop.IOLoop.current().start()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Start the ParlAI-MTurk task managing server.')
+    parser.add_argument('-port', metavar='port', type=int, default=DEFAULT_PORT,
+                        help='port to run the server on.')
+    parser.add_argument('-hostname', metavar='hostname', type=str,
+                        default=DEFAULT_HOSTNAME,
+                        help='host to run the server on.')
+    parser.add_argument('-db_file', metavar='db_file', type=str,
+                        default=DEFAULT_DB_FILE,
+                        help='name of database to use (in core/run_data)')
+    parser.add_argument('-logging_level', metavar='logger_level', default='INFO',
+                        help='logging level (default = INFO). Can take logging '
+                             'level name or int (example: 20)')
+    FLAGS = parser.parse_args()
+
+    try:
```

Comment:
i'm pretty sure a call to _checkLevel already typechecks for you, so you can just always call that (and you don't neeed this try/except block